doc/drop_counters/drop_counters_test_plan.md

```diff
@@ -0,0 +1,584 @@
+
+### 4.4.2. Verify that two drop counters with overlapping drop reasons both count drops correctly
+#### Counters:
+```
+TEST_COUNTER_0
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC
+
+TEST_COUNTER_1
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC, SIP_LINK_LOCAL, DIP_LINK_LOCAL
+```
+
+#### Sample Packets:
+Packet A:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+...
+```
+
+Packet B:
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps:
+Because counter B includes all the drop reasons, the total in counter B should be equal to the total number of drops. Counter A should only be equal to the number of SMAC_EQUALS_DMAC packets sent.
+
+## 4.5 Configuration
+
+### 4.5.1. Verify that a drop reason can be added to a counter that has already been installed
+#### Counters:
+```
+TEST_COUNTER
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC -> SMAC_EQUALS_DMAC, SIP_LINK_LOCAL
+```
+
+#### Sample Packets:
+Packet A:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+...
+```
+
+Packet B:
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps:
+We essentially want to repeat the general testing procedure twice for this test. The procedure is as follows:
+1. Send SMAC_EQUALS_DMAC packets to the device, verify that the counts are correct.
+2. Add SIP_LINK_LOCAL as a drop reason to the counter, verify that it is cleared.
+3. Send both SMAC_EQUALS_DMAC and SIP_LINK_LOCAL packets to the device, verify that both are counted by the drop counter.
+
+### 4.5.2. Verify that a drop reason can be removed from a counter that has already been installed
+#### Counters:
+```
+TEST_COUNTER
+    Type: INGRESS
+    Reasons: SMAC_EQUALS_DMAC, SIP_LINK_LOCAL -> SMAC_EQUALS_DMAC
+```
+
+#### Sample Packets:
+Packet A:
+```
+###[ Ethernet ]###
+  dst = [DUT_MAC_ADDR]
+  src = [DUT_MAC_ADDR]
+  type = 0x800
+...
+```
+
+Packet B:
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = 169.254.0.0
+    dst = 2.2.2.2
+...
+```
+
+#### Additional Steps:
+We essentially want to repeat the general testing procedure twice for this test. The procedure is as follows:
+1. Send both SMAC_EQUALS_DMAC and SIP_LINK_LOCAL packets to the device, verify that the counts are correct.
+2. Remove SIP_LINK_LOCAL as a drop reason from the counter, verify that it is cleared.
+3. Send both SMAC_EQUALS_DMAC and SIP_LINK_LOCAL packets to the device, verify that both are counted by RX_DRP but only SMAC_EQUALS_DMAC is counted by the drop counter.
```

Comment:
Can you please also clarify "counted by the drop counter". Because there is mentioned "RX_DRP" and "counted by the drop counter".

Reply:
Should be clarified now!